jme3-core/src/main/java/com/jme3/material/MatParamTexture.java

```diff
@@ -40,51 +40,83 @@
 import com.jme3.texture.image.ColorSpace;
 import java.io.IOException;
 
+/**
+ * A material parameter that holds a reference to a texture and its required color space.
+ * This class extends {@link MatParam} to provide texture specific functionalities.
+ */
 public class MatParamTexture extends MatParam {
 
-    private Texture texture;
     private ColorSpace colorSpace;
 
+    /**
+     * Constructs a new MatParamTexture instance with the specified type, name,
+     * texture, and color space.
+     *
+     * @param type       the type of the material parameter
+     * @param name       the name of the parameter
+     * @param texture    the texture associated with this parameter
+     * @param colorSpace the required color space for the texture
+     */
     public MatParamTexture(VarType type, String name, Texture texture, ColorSpace colorSpace) {
         super(type, name, texture);
-        this.texture = texture;
         this.colorSpace = colorSpace;
     }
 
+    /**
+     * Serialization only. Do not use.
+     */
     public MatParamTexture() {
     }
 
+    /**
+     * Retrieves the texture associated with this material parameter.
+     *
+     * @return the texture object
+     */
     public Texture getTextureValue() {
-        return texture;
+        return (Texture) getValue();
     }
 
+    /**
+     * Sets the texture associated with this material parameter.
+     *
+     * @param value the texture object to set
+     * @throws IllegalArgumentException if the provided value is not a {@link Texture}
+     */
     public void setTextureValue(Texture value) {
-        this.value = value;
-        this.texture = value;
+        setValue(value);
     }
     
+    /**
+     * Overrides the base class {@link MatParam#setValue(Object)} to allow null
+     * values.
+     *
+     * @param value the object to set as the value (must be a {@link Texture} or null)
+     * @throws IllegalArgumentException if the provided value is not a {@link Texture} and not null
+     */
     @Override
     public void setValue(Object value) {
-        if (!(value instanceof Texture)) {
-            throw new IllegalArgumentException("value must be a texture object");
+        if ((value == null) || value instanceof Texture) {
```

Comment:
seems to me this changes the behavior of `setValue(null)`. Was that your intention?

Reply:
yes, currently invoking the MatParamTexture.setValue(null) method raises an exception, while invoking the MatParam.setValue(null) method does not.

It seems reasonable to remove the setValue method from the MatParamTexture class.

The superclass, MatParam, already has a setValue method that performs all the required checks.
By removing the setValue method from MatParamTexture, it would inherit the behavior from the superclass, which aligns with the intended functionality.

Reply:
Go ahead and remove the overriding method. That seems the cleanest solution.

Reply:
Done